addons/website_crm/controllers/main.py

```diff
@@ -85,6 +85,13 @@ def insert_record(self, request, model, values, custom, meta=None):
             lang = request.context.get('lang', False)
             values['lang_id'] = values.get('lang_id') or request.env['res.lang']._lang_get_id(lang)
 
+            if 'email_from' in values:
+                partner = request.env['res.partner'].sudo().search([
```

Comment:
to triple check as discussed, we don't want people to be able to create task/lead for someone else's account (if there is a res user for that partner).